nixos/modules/services/networking/fedimintd.nix

```
@@ -0,0 +1,363 @@
+            ''
+          );
+        in
+        {
+          description = "Fedimint Server";
+          documentation = [ "https://github.com/fedimint/fedimint/" ];
+          wantedBy = [ "multi-user.target" ];
+          environment = lib.mkMerge [
+            {
+              FM_BIND_P2P = "${cfg.p2p.bind}:${toString cfg.p2p.port}";
+              FM_BIND_API = "${cfg.api.bind}:${toString cfg.api.port}";
+              FM_P2P_URL = cfg.p2p.url;
+              FM_API_URL = cfg.api.url;
+              FM_DATA_DIR = cfg.dataDir;
+              FM_BITCOIN_NETWORK = cfg.bitcoin.network;
+              FM_BITCOIN_RPC_URL = cfg.bitcoin.rpc.url;
+              FM_BITCOIN_RPC_KIND = cfg.bitcoin.rpc.kind;
+            }
+            cfg.environment
+          ];
+          serviceConfig = {
+            User = cfg.user;
+            Group = cfg.group;
+
+            StateDirectory = "fedimintd-${fedimintdName}";
+            StateDirectoryMode = "0700";
+            ExecStart = startScript;
+
+            Restart = "always";
+            RestartSec = 10;
+            StartLimitBurst = 5;
+            UMask = "007";
+            LimitNOFILE = "100000";
+
+            LockPersonality = true;
+            MemoryDenyWriteExecute = "true";
+            NoNewPrivileges = "true";
+            PrivateDevices = "true";
+            PrivateMounts = true;
+            PrivateTmp = "true";
+            ProtectClock = true;
+            ProtectControlGroups = true;
+            ProtectHostname = true;
+            ProtectKernelLogs = true;
+            ProtectKernelModules = true;
+            ProtectKernelTunables = true;
+            ProtectSystem = "full";
+            RestrictAddressFamilies = [
+              "AF_INET"
+              "AF_INET6"
+            ];
+            RestrictNamespaces = true;
+            RestrictRealtime = true;
+            SystemCallArchitectures = "native";
+            SystemCallFilter = [
+              "@system-service"
+              "~@privileged"
+            ];
+          };
+        }
+      ))
+    ) eachFedimintd;
+
+    users.users = mapAttrs' (
+      fedimintdName: cfg:
+      (nameValuePair "fedimintd-${fedimintdName}" {
+        name = cfg.user;
+        group = cfg.group;
+        description = "Fedimint daemon user";
+        home = cfg.dataDir;
+        isSystemUser = true;
+      })
+    ) eachFedimintd;
+
+    users.groups = mapAttrs' (fedimintdName: cfg: (nameValuePair "${cfg.group}" { })) eachFedimintd;
+
+    services.nginx.virtualHosts = mapAttrs' (
+      fedimintdName: cfg:
+      (nameValuePair cfg.api.fqdn (
+        lib.mkMerge [
+          cfg.nginx.config
+
+          {
+            enableACME = mkDefault true;
+            forceSSL = mkDefault true;
+            # Currently Fedimint API only support JsonRPC on `/ws/` endpoint, so no need to handle `/`
+            locations."/ws/" = {
+              proxyPass = "http://127.0.0.1:${builtins.toString cfg.api.port}/";
```

Small nit, not blocking

```suggestion
              proxyPass = "http://127.0.0.1:${toString cfg.api.port}/";
```